### PR TITLE
Adding automation around ADR authoring

### PR DESCRIPTION
**Pull request by rogeruiz**
This patch adds support for running `npm run new-adr --slug="my-new-adr"` and
have it produce a new ADR file from the template with the suggested front-matter
from the documentation on the ADR guide page. This will help folks contributing
ADRs in the correct way without a whole lot of cognitive overhead.

### CODE REVIEW

**Review by YanZ777 — APPROVED** [review]
Tried it out and works for me!

File: scripts/new-adr.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+num_prefix=$(printf '%04d' "$(find docs/adrs -iname '*.md' | wc -l)")
+
+new_adr_path="./docs/adrs/${num_prefix}-${slug}.md"
+
+cat << ADRHD > "${new_adr_path}"
+---
+title: '${num_prefix} A title for this ADR'
+description: |
+  A description for ADR ${num_prefix}
```
Comment (felipe-lee) [inline scripts/new-adr.sh:27]: **suggestion (non-blocking)**: This doesn't seem to be part of the regular template, but I wonder if we should add it, for anyone that doesn't use this script. I would say that could be a separate PR except that would affect the `tail` command below, so it might be good to do it all in one PR.

**Review by felipe-lee — APPROVED** [review]
LGTM! Just left a comment, but I'm good with this whether or not that is taken into account.

File: scripts/new-adr.sh (hunk 1 above)
Comment (rogeruiz) [reply scripts/new-adr.sh:27]: Thanks for the suggestion @felipe-lee I'll add it to this PR.
